packages/flutter/test/widgets/focus_manager_test.dart

```diff
@@ -785,7 +785,7 @@ void main() {
       await tester.pump();
       expect(tester.binding.focusManager.primaryFocus, isNot(equals(child2)));
       expect(tester.binding.focusManager.primaryFocus, isNot(equals(child1)));
-      expect(scope.focusedChild, equals(child1));
+      expect(scope.focusedChild, isNull);
```

Comment:
It looks like the test was originally set up to expect `isNull` but then was changed in #50831.

I don't exactly know the intent behind the change; hopefully @gspencergoog can weigh in.

Reply:
This is because the PR appears to clear out the list of focused children more aggressively than it should.  If `focusedChild` on the scope is not equal to `child1` here, then when the scope is refocused later, `child1` won't get focus unless it also happens to be the first focus for the scope. The `unfocus` method hasn't been called here explicitly.

That change (#50831) fixed this behavior so that it would work this way, which was the intended behavior.